src/main/java/org/mockito/internal/matchers/Equals.java

```diff
@@ -55,8 +55,18 @@ public String toStringWithType() {
         return "(" + wanted.getClass().getSimpleName() + ") " + describe(wanted);
     }
 
+    @Override
+    public String toStringWithFullName() {
```

Comment:
I am not the biggest fan of adding another method here. Especially as it is very similar to the `toStringWithType()`, but the only difference is the name retrieval.

What if instead we get the wanted object and perform the computation in the `MatchersPrinter`? E.g. avoid introductions of new methods and pass in the desired name into `toStringWithType(String className)`